News-Android-App/src/main/java/de/luhmer/owncloudnewsreader/NewsReaderDetailFragment.java

```diff
@@ -338,6 +352,86 @@ protected void onPostExecute(List<RssItem> rssItem) {
         }
     }
 
+    private class SearchInCurrentRssViewTask extends UpdateCurrentRssViewTask {
+
+        final private String searchString;
+
+        /**
+         * Display only items in current view, where Title OR Body contain the searchString
+         * @param context Context
+         * @param searchString String to search for
+         */
+        public SearchInCurrentRssViewTask(final Context context, final String searchString)
+        {
+            super(context);
+            this.searchString=searchString;
+        }
+
+        @Override
+        protected List<RssItem> doInBackground(Void... urls) {
+            SharedPreferences mPrefs = PreferenceManager.getDefaultSharedPreferences(super.getContext());
+            DatabaseConnectionOrm dbConn = new DatabaseConnectionOrm(super.getContext());
+            boolean onlyUnreadItems = mPrefs.getBoolean(SettingsActivity.CB_SHOWONLYUNREAD_STRING, false);
+            boolean onlyStarredItems = false;
+
+            String sqlSelectStatement = null;
+            if (idFeed != null) {
+                sqlSelectStatement = getFeedSQLStatement(idFeed, onlyUnreadItems, onlyStarredItems, sortDirection, searchString, dbConn, mPrefs);
+
+            } else if (idFolder != null) {
+                if (idFolder == SubscriptionExpandableListAdapter.SPECIAL_FOLDERS.ALL_STARRED_ITEMS.getValue()) {
+                    onlyUnreadItems = false;
+                }
+                sqlSelectStatement = getFolderSQLStatement(idFolder, onlyUnreadItems, sortDirection,searchString, dbConn, mPrefs);
+            }
+
+            List<RssItem> items = null;
+            if (sqlSelectStatement != "") {
+                dbConn.insertIntoRssCurrentViewTable(sqlSelectStatement);
+                items = dbConn.getCurrentRssItemView(0);
+            }
+            return items;
+
+        }
+
+        private String getFeedSQLStatement(final long idFeed,
+                                           final boolean onlyUnreadItems,
+                                           final boolean onlyStarredItems,
+                                           final SORT_DIRECTION sort_direction,
+                                           final String searchString,
+                                           final DatabaseConnectionOrm dbConn,
+                                           final SharedPreferences mPrefs)
+        {
+            String sql="";
+            String searchIn = mPrefs.getString(SettingsActivity.SP_SEARCH_IN,"0");
+            if(searchIn.equals("0")) {
```

Comment:
`"0"`and `"1"` should be put into constants, see same thing for `getFolderSQLStatement` since 0 and 1 have a meaning this having a speaking name for them makes it easier to read the code :)